connectors/sources/gmail.py

```diff
@@ -0,0 +1,354 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import json
+from functools import cached_property
+
+import fastjsonschema
+from fastjsonschema import JsonSchemaValueException
+
+from connectors.filtering.validation import (
+    AdvancedRulesValidator,
+    SyncRuleValidationResult,
+)
+from connectors.source import BaseDataSource, ConfigurableFieldValueError
+from connectors.sources.google import (
+    GMailClient,
+    GoogleDirectoryClient,
+    MessageFields,
+    UserFields,
+)
+from connectors.utils import base64url_to_base64, iso_utc
+
+ACCESS_CONTROL = "_allow_access_control"
```

Comment:
Should this go to `utils`?

Reply:
Yes, tracked through [Add common access control module](https://github.com/elastic/enterprise-search-team/issues/5594)